File: java/solution/src/main/java/Webshop/Service/Products/ProductRepository.java
package Webshop.Service.Products;

import java.util.HashMap;
import java.util.Map;

public class ProductRepository implements IProductRepository {
    private final Map<String, Product> productDatabase = new HashMap<>();
    private final Map<String, Integer> inventoryDatabase = new HashMap<>();

    @Override
    public Product getProduct(String id) {
        return productDatabase.get(id);
    }

    @Override
    public int getProductInventory(String productId) {
        return inventoryDatabase.getOrDefault(productId, 0);
    }

    @Override
    public void updateProductInventory(String productId, int newInventory) {
        inventoryDatabase.put(productId, newInventory);
    }
}

interface IProductRepository {
    Product getProduct(String id);
    int getProductInventory(String productId);
    void updateProductInventory(String productId, int newInventory);
}

File: csharp/exercises/Webshop.Service/Products/ProductRepository.cs
﻿namespace Webshop.Service.Products;

public class ProductRepository : IProductRepository
{
    private Dictionary<string, Product> _productDatabase = new Dictionary<string, Product>();
    private Dictionary<string, int> _inventoryDatabase = new Dictionary<string, int>();

    public Product? GetProduct(string id)
    {
        if (_productDatabase.TryGetValue(id, out var product))
            return product;
        return null;
    }

    public int GetProductInventory(string productId)
    {
        if (_inventoryDatabase.TryGetValue(productId, out var productInventory))
            return productInventory;
        return 0;
    }

    public void UpdateProductInventory(string productId, int newInventory)
    {
        _inventoryDatabase[productId] = newInventory;
    }
}

public interface IProductRepository
{
    Product? GetProduct(string id);
    int GetProductInventory(string productId);
    void UpdateProductInventory(string productId, int newInventory);
}

public class Product
{
    public string Id { get; set; }
    public decimal Price { get; set; }
}